examples/static_meshlets/source/static_meshlets.cpp

```diff
@@ -0,0 +1,497 @@
+#endif
+
+		glm::mat4 mModelMatrix;
+
+		int32_t mMaterialIndex;
+	};
+
+	struct loaded_data_for_draw_call
+	{
+		std::vector<glm::vec3> mPositions;
+		std::vector<glm::vec2> mTexCoords;
+		std::vector<glm::vec3> mNormals;
+		std::vector<uint32_t> mIndices;
+#if !USE_DIRECT_MESHLET
+		std::vector<uint32_t> mMeshletData;
+#endif
+
+		glm::mat4 mModelMatrix;
+
+		int32_t mMaterialIndex;
+	};
+
+	struct alignas(16) meshlet
+	{
+		glm::mat4 mTransformationMatrix;
+		uint32_t mMaterialIndex;
+		uint32_t mTexelBufferIndex;
+
+#if USE_DIRECT_MESHLET
+		gvk::meshlet_gpu_data<sNumVertices, sNumIndices> mGeometry;
+#else
+		gvk::meshlet_redirected_gpu_data mGeometry;
+#endif
+	};
+
+public: // v== avk::invokee overrides which will be invoked by the framework ==v
+	static_meshlets_app(avk::queue& aQueue)
+		: mQueue{ &aQueue }
+	{}
+
+	void initialize() override
+	{
+		// use helper functions to create ImGui elements
+		auto surfaceCap = gvk::context().physical_device().getSurfaceCapabilitiesKHR(gvk::context().main_window()->surface());
+
+		// Create a descriptor cache that helps us to conveniently create descriptor sets:
+		mDescriptorCache = gvk::context().create_descriptor_cache();
+
+		glm::mat4 globalTransform = glm::scale(glm::vec3(0.01f));
+		std::vector<gvk::model> loadedModels;
+		// Load a model from file:
+		auto sponza = gvk::model_t::load_from_file("assets/sponza_structure.obj", aiProcess_Triangulate | aiProcess_PreTransformVertices);
+
+		loadedModels.push_back(std::move(sponza));
+
+
+		std::vector<gvk::material_config> allMatConfigs; // <-- Gather the material config from all models to be loaded
+		std::vector<std::vector<glm::mat4>> meshRootMatricesPerModel;
+		std::vector<loaded_data_for_draw_call> dataForDrawCall;
+		std::vector<meshlet> meshletsGeometry;
+
+
+		for (size_t i = 0; i < loadedModels.size(); ++i) {
+			auto curModel = std::move(loadedModels[i]);
+
+			// get all the meshlet indices of the model
+			const auto meshIndicesInOrder = curModel->select_all_meshes();
+
+			auto distinctMaterials = curModel->distinct_material_configs();
+			const auto matOffset = allMatConfigs.size();
+			// add all the materials of the model
+			for (auto& pair : distinctMaterials) {
+				allMatConfigs.push_back(pair.first);
+			}
+
+			for (size_t mpos = 0; mpos < meshIndicesInOrder.size(); mpos++) {
+				auto meshIndex = meshIndicesInOrder[mpos];
+				std::string meshname = curModel->name_of_mesh(mpos);
+
+				auto texelBufferIndex = dataForDrawCall.size();
+				auto& drawCallData = dataForDrawCall.emplace_back();
+
+				drawCallData.mMaterialIndex = static_cast<int32_t>(matOffset);
+				drawCallData.mModelMatrix = globalTransform * curModel->transformation_matrix_for_mesh(meshIndex);
+				// Find and assign the correct material (in the ~"global" allMatConfigs vector!)
+				for (auto pair : distinctMaterials) {
+					if (std::end(pair.second) != std::find(std::begin(pair.second), std::end(pair.second), meshIndex)) {
```

Comment:
```suggestion
					if (std::end(pair.second) != std::ranges::find(std::begin(pair.second), std::end(pair.second), meshIndex)) {
```

I think, we could make use of `std::ranges` also here.